src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -840,6 +857,21 @@ internal async Task InvokeResponse(InvocationResponse invokeResponse)
                         SendCloseSharedMemoryResourcesForInvocationRequest(outputMaps);
                     }
                 }
+
+                string data = $"WorkerId: {Id}, TotalInvocations : {_invocationMetricsPerWorkerId[Id].TotalInvocations}, SuccessfulInvocations : {_invocationMetricsPerWorkerId[Id].SuccessfulInvocations}, AverageInvocationLatency : {_invocationMetricsPerWorkerId[Id].AverageInvocationLatency}";
+                _metricsLogger.LogEvent(string.Format(MetricEventNames.WorkerInvocationStatus, Id), data: data);
```

Comment:
Regarding the data payload on these events. If you look at LogEvent [here](https://github.com/Azure/azure-functions-host/blob/dcfd7fc1458da020de2c3f49b1d9672b1e1bc229/src/WebJobs.Script.WebHost/Diagnostics/MetricsEventManager.cs#L118) you'll see that aggregations are done based on event name / function name only, so in this case the resulting data from the aggregated event will be from the first event. It's confusing and funky but that's the way it works - for a particular event, if data is being used, that data should be constant for all the events.

Reply:
Corrected code here